torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1937,52 +1937,54 @@ std::vector<Val*> Index::getNonGlobalProducerStridedIndices(
   return strided_inds;
 }
 
-std::vector<Val*> Index::getLinearIndex(
-    TensorView* consumer_tv,
-    const std::vector<kir::ForLoop*>& loops) {
+template <typename func_t>
```

Comment:
Doesn't need to be in this PR. But this function looks more like a domain guard factory.